api/graphql/schema.graphql

```diff
@@ -181,6 +181,9 @@ type Pool {
     @hasMany(relation: "publishedPoolCandidates", scopes: ["authorizedToView"])
     @canResolved(ability: "view")
     @guard
+  poolCandidatesCount: Int
+    @count(relation: "poolCandidates", scopes: ["notDraft"])
+    @guard
```

Comment:
Designers are fine with this being public info, so we can even remove the `@guard` directive!

Reply:
I was probably a bit wrong on the slowness, I think policy checking probably also had a big part in cratering performance

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/038a41c4-d888-4d9f-b44a-89d3d91643bf)

For the purposes of counting, probably overkill to policy check each candidate of a possible thousand(s).
Important when doing things other than counting

Reply:
Yeah, policy checks really slow down list queries since it needs to check each entity individually :cry: